src/App.tsx

```diff
@@ -15,36 +17,109 @@
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+const SORT_FIELD_ALPHABET: string = 'alphabet';
+const SORT_FIELD_LENGTH: string = 'length';
+const SORT_FIELD_RESET: string = '';
+
+type Props = {
+  sortField: string;
+  setSortField: () => string;
+  isReverse: string;
+  setReverse: () => string;
+  visibleGoods: string[];
+};
+
+function sortGoods(
+  goods: string[],
+  sortField: string = '',
+  isReverse: boolean = false,
+): string[] {
+  const visibleGoods: string[] = [...goods];
+
+  if (sortField) {
+    switch (sortField) {
+      case SORT_FIELD_ALPHABET:
+        visibleGoods.sort((good1, good2) => good1.localeCompare(good2));
+        break;
+      case SORT_FIELD_LENGTH:
+        visibleGoods.sort((good1, good2) => good1.length - good2.length);
+        break;
+      default:
+        throw new Error('Sort method is unknown!');
+    }
+  }
+
+  if (isReverse) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
+export const App: React.FC<Props> = () => {
```

Comment:
[CODE STYLE] - The name of the component `App` should be more descriptive of its functionality. Consider renaming it to better reflect what it represents or does, e.g., `GoodsSorter` or `GoodsList`.